src/main/java/reactor/rabbitmq/OutboundMessageResult.java

```diff
@@ -19,9 +19,9 @@
 /**
  * Result of a sent message when using publisher confirm.
  */
-public class OutboundMessageResult {
+public class OutboundMessageResult<T> {
```

Comment:
similarly to the above comment, I'd extract this specificity in a `OutboundCorrelatedMessageResult<T>`. Then `Sender` can have methods that take a publisher of `OutboundMessage` and return `Flux<OutboundMessageResult` vs methods that take a publisher of `OutboundCorrelatedMessage<T>` and return `Flux<OutboundCorrelatedMessageResult<T>>`